### PR TITLE
add hook args lookup resolution with a test and doc updates

### PR DESCRIPTION
**Pull request by ITProKyle**
implementation from cloudtools/stacker#708

### CODE REVIEW

File: runway/cfngin/hooks/utils.py [hunk 1]
```diff
@@ -5,6 +5,11 @@
 import sys
 
 from runway.util import load_object_from_string
+# from runway.variables import Variable, resolve_variables
+
+from ..exceptions import FailedVariableLookup
+# TODO replace after https://github.com/onicagroup/runway/pull/116
```
Comment (troyready) [inline runway/cfngin/hooks/utils.py]: @ITProKyle merged ^ and ready for updating

Comment (ITProKyle) [reply runway/cfngin/hooks/utils.py]: updated

**Comment by troyready** [comment]
Not sure what to make of the disttools error in travis -- is it apparent to you?

**Comment by ITProKyle** [comment]
looking into it. nothing changed for us on that front so possibly a new version of something broke it.

**Comment by ITProKyle** [comment]
@troyready - fixed the build error and an issue I found with the executables output by Pyinstaller in #145

**Pull request merged**